gtsfm/utils/io.py

```diff
@@ -153,6 +191,34 @@ def write_cameras(gtsfm_data: GtsfmData, images: List[Image], save_dir: str) ->
             f.write(f"{i} {camera_model} {image_width} {image_height} {fx} {u0} {v0} {k1} {k2}\n")
 
 
+def read_images_txt(fpath: str) -> Tuple[List[Pose3], List[str]]:
+    """Read camera poses and image file names from a COLMAP-format images.txt file.
+
+    Reference: https://colmap.github.io/format.html#images-txt
+
+    Args:
+        fpath: path to images.txt file
+
+    Returns:
+        wTi_list: list of camera poses for each image
+        img_fnames: name of image file, for each image
+    """
+    with open(fpath, "r") as f:
+        lines = f.readlines()
+
+    wTi_list = []
+    img_fnames = []
+    # ignore first 4 lines of text -- they are a description of the file format
+    for line in lines[4::2]:
+        i, qw, qx, qy, qz, tx, ty, tz, i, img_fname = line.split()
+        wRi = Rot3(float(qw), float(qx), float(qy), float(qz))
+        wTi = Pose3(wRi, np.array([float(tx),float(ty),float(tz)]))
```

Comment:
`np.array([tx, ty, tz], dtype=np.float64)`?

Reply:
thanks -- yes this works and is cleaner

Reply:
updated